app.py

```diff
@@ -205,13 +206,80 @@ def get_service_data(service_id):
     return ServiceService.get_service(service_id)    
 
 @app.route('/invoices/add', methods=["POST"])
-def add_new_invoice():
+def add_new_estimate():
     if not g.user:
         flash("Access unauthorized.", "danger")
         return redirect("/")
+    cust_id = json.loads(request.data)['customerId']
+    print(cust_id)
+    comp_id = g.user.employer.company_id  
+    
+    estimate = InvoiceService.create_estimate(cust_id=cust_id, comp_id=comp_id)
+
 
-    print(json.loads(request.data))
 
-    service_ids = json.loads(request.data['services'])
-    # for s_id in service_ids:
+    services_and_quantity = json.loads(request.data)['services']
+    for service in services_and_quantity:
+        ServiceRequestService.add_service_request(
+                                service_id=service['serviceId'], 
+                                quantity=service['quantity'], 
+                                invoice_id=estimate.id)
+    return {'id': estimate.id}
 
+@app.route('/invoices/finalize/<int:estimate_id>', methods=["POST"])
+def finalize_invoice(estimate_id):
+    if not g.user:
+        flash("Access unauthorized.", "danger")
+        return redirect("/")
+    invoice = InvoiceService.get_invoice_by_id(estimate_id)
+    if g.user.employer.company_id != invoice.company_id:
```

Comment:
Let's pull the permission logic into a decorator that you can decorate any routes